test/specs/controller.spec.js

```diff
@@ -70,4 +70,110 @@ describe('controller', function() {
     expect(so).toEqual(jasmine.objectContaining({x: 2}));
     expect(ag).toEqual(jasmine.objectContaining({x: 3}));
   });
+
+  it('should parse a complex flow', function() {
```

Comment:
Also could you split this and the fix (not sure what it is currently) for it in separate PR?